content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -185,6 +185,50 @@ echo useSomeLib(new Helper('some text'))
 
 Global Variables however, are resolved at runtime.
 
+=== Loading libraries dynamically
+
+As of version 2.7 of the _Pipeline: Shared Groovy Libraries_ plugin,
+there is a new option for loading (non-implicit) libraries in a script:
+a `library` step that loads a library _dynamically_, at any time during the build.
+
+If you are only interested in using global variables/functions (from the `vars/` directory),
+the syntax is quite simple:
```

Comment:
I think it would be very helpful to include some of the reasons why one might want to use `library` versus the `@Library` annotation. IIRC this helps unlock some functionality in Declarative Pipeline? Perhaps a small contrived `Jenkinsfile` demonstrating the "appropriate scenario" in which to use the `library` step.


I want to make sure that users (myself included) are using the step in a manner in which it was intended.